apps/site/assets/ts/schedule/schedule-loader.tsx

```diff
@@ -27,40 +32,60 @@ const renderMap = ({
   );
 };
 
+const updateURL = (direction: DirectionId, origin: SelectedOrigin): void => {
+  if (window) {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    const newQuery = {
+      "schedule_finder[direction_id]": direction.toString(),
+      "schedule_finder[origin]": origin
+    };
+    const newLoc = updateInLocation(newQuery, window.location);
+    // newLoc is not a true Location, so toString doesn't work
+    window.history.replaceState({}, "", `${newLoc.pathname}${newLoc.search}`);
+  }
+};
+
 const renderSchedulePage = (schedulePageData: SchedulePageData): void => {
+  const {
+    direction_id: directionId,
+    schedule_note: scheduleNote
+  } = schedulePageData;
+
   ReactDOM.render(
-    <SchedulePage schedulePageData={schedulePageData} />,
+    <Provider store={store}>
+      <ScheduleLoader
+        component="MAIN"
+        schedulePageData={schedulePageData}
+        selectedDirection={directionId}
+        updateURL={updateURL}
+      />
+    </Provider>,
     document.getElementById("react-root")
   );
-  if (schedulePageData.schedule_note) {
+  // Now we (conditionally) render "secondary" widgets:
+  // (only one will be shown at a time, depending on whether mobile or desktop)
+  if (scheduleNote) {
     ReactDOM.render(
-      <ScheduleNote
-        className="m-schedule-page__schedule-notes--mobile"
-        scheduleNote={schedulePageData.schedule_note}
-      />,
+      <Provider store={store}>
+        <ScheduleLoader
+          component="SCHEDULE_NOTE"
+          schedulePageData={schedulePageData}
+          selectedDirection={directionId}
+          updateURL={updateURL}
+        />
+      </Provider>,
       document.getElementById("react-schedule-note-root")
     );
-  }
-  const {
-    direction_id: directionId,
-    route,
-    stops,
-    services,
-    route_patterns: routePatternsByDirection,
-    schedule_note: scheduleNote,
-    today
-  } = schedulePageData;
-  if (!scheduleNote) {
+  } else {
     ReactDOM.render(
-      <ScheduleFinder
-        directionId={directionId}
-        route={route}
-        stops={stops}
-        services={services}
-        routePatternsByDirection={routePatternsByDirection}
-        today={today}
-        scheduleNote={null}
-      />,
+      <Provider store={store}>
```

Comment:
@alicia-mbta not too familiar with `Provider`s, what's this about?

Reply:
@phildarnowsky it's a redux thing! it's what enables the `ScheduleLoader` to connect to the store